opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -356,12 +356,48 @@ def __init__(
     ):
         self.instrumentation_info = instrumentation_info
         self.processor = Processor(source.stateful, source.resource)
-        self.metrics = set()
-        self.observers = set()
+        self.metrics = dict()
+        self.observers = dict()
```

Comment:
nit
```suggestion
        self.metrics = {}
        self.observers = {}
```

Reply:
Today I Learned: `{}` is faster than `dict()`. Thanks.